### ISSUE TITLE
staticcheck: detect duplicated conditionals

### ISSUE THREAD

**Issue by ainar-g** [dominikh/go-tools#193]
I've noticed this bug in some of the code I work with. The original code deals with database transactions, but the simplified version is this: https://play.golang.org/p/5FJy6AVZ2E. This code contains a subtle bug:

    errR := rollback()
    if err != nil { // WRONG
        log.Printf("rollback error: %v", errR)
    }

should be

    errR := rollback()
    if errR != nil { // RIGHT
        log.Printf("rollback error: %v", errR)
    }

A diagnostic could see that there is an `if err != nil` condition within an `if err != nil` condition, which will always be true, and raise an error.

This might be a bug in SA4014, but I'm not sure.

**Comment by shurcooL** [comment]
> see that there is an `if err != nil` condition within an `if err != nil` condition, which will always be true

As long as `err` is not modified between the two checks.

**Comment by ainar-g** [comment]
>As long as `err` is not modified between the two checks.

Yes, this check would obviously need to take side effects into account.

**Comment by ainar-g** [comment]
If that helps, [here](https://github.com/llvm-mirror/clang/blob/a68f40374973e5ea579333e61f6ebbf425c78f09/lib/StaticAnalyzer/Checkers/IdenticalExprChecker.cpp#L107) is the source for a similar check in clang-analyzer.

**Comment by dominikh** [comment]
Thanks for the link. The implementation is indeed not very challenging (at least for local variables), and it's on my todo list. But that is a very long list.
